.github/workflows/branch-push.yaml

```diff
@@ -0,0 +1,60 @@
+name: CI pipeline
+
+on:
+  pull_request:
+    branches:
+      - "**"  # Runs on all pull requests
+
+jobs:
+  check-test-coverage:
+    # The coverage is configured inside 'jest.config.js' by setting the 'coverageThreshold'
+    name: Checking test coverage (must be over 90%)
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        node-version: [ 20.17 ]  # Define Node.js latest LTS version (update as needed)
+
+    steps:
+      # Step 1: Checkout the code
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      # Step 2: Set up Node.js with the required version
+      - name: Set up Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: ${{ matrix.node-version }}
+
+      # Step 3: Install dependencies
+      - name: Install dependencies
+        run: npm install
+
+      # Step 4: Run Jest tests
+      - name: Run Jest tests
+        # TODO: Adapt to run coverage on changed files only.
+        # run: npm test -- --coverage # $(git diff --name-only origin/main...HEAD | grep '\.[jt]sx\?$' | grep -v '\ .test\.')
```

Comment:
see above. I guess we can leave this out.
If we need it you could reaise an issue and delete this line anyway :)